news-bundle/src/Resources/contao/modules/ModuleNews.php

```diff
@@ -98,12 +98,45 @@ protected function parseArticle($objArticle, $blnAddArchive=false, $strClass='',
 		$objTemplate->hasTeaser = false;
 		$objTemplate->hasReader = true;
 
-		// Clean the RTE output
-		if ($objArticle->teaser)
+		switch ($objArticle->teaserType)
 		{
-			$objTemplate->hasTeaser = true;
-			$objTemplate->teaser = StringUtil::toHtml5($objArticle->teaser);
-			$objTemplate->teaser = StringUtil::encodeEmail($objTemplate->teaser);
+			case 'elements':
+				$objTemplate->hasTeaser = static function () use ($objArticle)
+				{
+					return News::hasNewsContent($objArticle);
+				};
+
+				$objTemplate->teaser = static function () use ($objArticle)
+				{
+					return News::getNewsContent($objArticle, ['limit' => $objArticle->teaserElements]);
+				};
+				break;
+
+			case 'substr':
+				$objTemplate->hasTeaser = static function () use ($objArticle)
+				{
+					return News::hasNewsContent($objArticle);
+				};
+
+				$objTemplate->teaser = static function () use ($objArticle)
+				{
+					return StringUtil::substrHtml(
+						News::getNewsContent($objArticle),
+						$objArticle->teaserChars
+					);
```

Comment:
I don’t think `StringUtil::substrHtml()` is reliable for large HTML chunks (that might include script tags e.g.).